### PR TITLE
User interface

### PR DESCRIPTION
**Pull request by icebox827**
# Milestone 2
- Merge the previous feature branch
- Create the toke variable
- Create  the board variable
- Create a method to ask the player for names and assign token
- Create a method to display the board
- Create a method to ask the player for moves and play the game

### CODE REVIEW

**Review by NtwaliHeritier — CHANGES_REQUESTED** [review]
**CHANGES REQUIRED**
Hello @icebox827 @oxenprogrammer , Great work so far, I can see you put a lot of effort in making this project. However some changes are required.

- When I chose an invalid key, the game should prompt me to choose another key. 
- Console should display the name of the player whose turn it is to play.
- The game board should display after each move.
- The console should show the list of available positions on the game board.
- Player a should not be allowed to pick the position already picked by player b or vice versa.
[REQUIRED] game tells which player turn it is
[REQUIRED] game displays board after the player move
[REQUIRED] game asks the player to select from available moves
[REQUIRED] game informs player if the selected move is invalid
[REQUIRED] game informs player if the selected move is a winning move
[REQUIRED] game informs player if the selected move is a draw move

Make the required changes and ask for another review. Feel free to contact me on slack(username: NTWALI Heritier)

File: bin/main.rb [hunk 1]
```diff
@@ -1,2 +1,66 @@
...
+      counter -= 1
+    else
+      move
+    end
+    counter += 1
+  end
+end
+
+# Call the methods
+greetings(tokens)
```
Comment (MarilenaRoque) [inline bin/main.rb]: - Argument not defined in the method

File: bin/main.rb [hunk 2]
```diff
@@ -1,2 +1,66 @@
...
+board_display(board)
+player_move
```
Comment (MarilenaRoque) [inline bin/main.rb]: - Wrong number of arguments

**Review by MarilenaRoque — CHANGES_REQUESTED** [review]
### Hello @icebox82! 🚀 

You already did a great job implementing the changes requested so far.  👏🏻 
However, your code is presenting some errors when running and some changes are still required to complete the requirements listed by my colleague.

## Changes Required ♻️ 

- [x] **Your code is presenting some error caused by calling methods with the wrong number of arguments.**
- [x] game tells which player turn it is
  - In each loop run, you should switch between players and display who's turn it is
- [x] game displays board after the player move
   - The method for display the board is working just fine, but make sure that it will work in every run of the loop.

- [x] game informs player if the selected move is a winning move
- [x] game informs player if the selected move is a draw move

  - Please fake a check(no logic is required yet) and build messages that will display if there's winner or a draw. Remember to check it for each move.
  
I am adding some helpful comments below. Please check it.

Feel free to contact me in Slack if you need flutter explanations.
Happy Coding! 😃 
 **_Marilena Roque
Full-Stack Developer
Code Reviewer at Microverse Inc.
[Portfolio](https://mariroqueproject.com/) | [Linkedin](https://www.linkedin.com/in/roquemarilena/) 
marilenas@id.uf.br_**

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGE REQUIRED :recycle: 
Hello team!! :clap: Great job working on the required changes.
However, there are some problems that you have to fix.
---------------------------
- [ ] I can choose a move that already is taken. It does not tell me if it is an invalid move. 
- [ ] [OPTIONAL] In the choose token part implement the downcase and uppercase so you can use either x or X and o or O. 

-------------------
Make the changes and request another review. :muscle: 

HAPPY CODING :smiley: 

MB5W4D

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED  :stop_sign: 
### HELLO  **TEAM**  I really like your project and you did a great job implementing the changes from the last review :+1:  very nice work :+1:  but unfortunately as much as I want to accept this pull request I can't because there is some error which is not letting me continue the game:
- [x] bin/main.rb:72:in `<main>': undefined method `**board_display**' for main:Object (NoMethodError). Kindly fix this

### **Please make the required changes and submit another review request when you are ready.**

<hr>
### if you have any question kindly reach me on slack @GzimAsani

### HAPPY CODING :computer:

**Review by Temmarie — CHANGES_REQUESTED** [review]
## Changes Required! :recycle:

### Hi @icebox827   ! :wave: :wave:

#### You have done well so far :clap: However, there are changes required for you to make in order for this project to be approved. Kindly work on them and submit them for another code review.


**Recommended**

- [ ] The positions picked are not registered on the board, please replace the positions with the players' tokens.(OPTIONAL)
- [ ] Game tells which player turn it is (without computing who the winner is)

- I can choose a move that already is taken. It does not tell me if it is an invalid move.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

**Review by LamiaSristy — CHANGES_REQUESTED** [review]
# Status: Change Required. 🔴 

Hello Team, Great work so far, 🙌 .
You have done all the changes, but there is a major fault in Please check my points here: 👇

- [ ] A loop needs to be terminated at some point (after a winning move or draw to be more specific). It can be done with a variable.

Once you make the changes, please ask again for a re-review.

Feel free to reach me at @lamiasristy 🙋

**Review by kevcap — APPROVED** [review]
# [STATUS] PROJECT APPROVED 🍾

### I really like your project! Congrats, you did an excellent job, you have my approval to go ahead and merge this pull request!

## As always, I wish you the very best and see you at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

**Pull request merged**